### PR TITLE
remove unused-parameter warnings in C++

### PR DESCRIPTION
**Pull request by chrisse74**
As described in  #16542 
"C++ code generated for streaming methods contains functions with unused arguments that trip up compilation with -Wunused-parameter (part of -Wextra)."

According to the Google C++ Style Guide ( section: Function_Declarations_and_Definitions ) this should be adressed by commenting out the unused parameter names.

This pull request handles all unused-parameter warnings, that I could find.

### CODE REVIEW

File: src/compiler/cpp_generator.cc [hunk 1]
```diff
@@ -1164,8 +1164,8 @@ void PrintHeaderServerMethodStreamedUnary(
         *vars,
         "// disable regular version of this method\n"
         "::grpc::Status $Method$("
-        "::grpc::ServerContext* context, const $Request$* request, "
-        "$Response$* response) override {\n"
+        "::grpc::ServerContext* /* context */, const $Request$* /* request */, "
```
Comment (sheenaqotj) [inline src/compiler/cpp_generator.cc]: Thanks for the changes. 

Please make whitespace consistent across all of these. Some of the changes look like "/* context */" and some look like "/*context*/". I prefer the latter.

**Comment by chrisse74** [comment]
Thanks for the review!

I removed the whitespace as suggested.

**Comment by sheenaqotj** [comment]
Please squash commits. Thanks again for the changes!

**Comment by hcaseyal** [comment]
@chrisse74 please resolve the test failures. tools/distrib/clang_format_code.sh should fix the SanityChecks errors.

**Comment by chrisse74** [comment]
something went wrong here (by squashing my commits) - the commit history differs from the intended one. sorry - guess i have to file a new pull request...

**Comment by chrisse74** [comment]
I will file a new pull request.

**Pull request closed without merging**